src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultMetadataDetails.cs

```diff
@@ -68,5 +68,10 @@ public DefaultMetadataDetails(ModelMetadataIdentity key, ModelAttributes attribu
         /// Gets or sets the <see cref="Metadata.ValidationMetadata"/>
         /// </summary>
         public ValidationMetadata ValidationMetadata { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ModelMetadata"/> of the container type.
+        /// </summary>
+        public ModelMetadata ContainerMetadata { get; set; }
```

Comment:
YAGNI. MMI is a struct, and it would be great not to make it bigger if possible.